## Offline Lidar VTR3
session_name: vtr_offline_velodyne_grizzly

#environment:
  #CYCLONEDDS_URI: ${VTRSRC}/config/cyclonedds.default.xml # custom dds configs
  # ROS_DOMAIN_ID: "1"  # set this to a unique number when multiple ROS2 dependent system running on the same network

start_directory: ${VTRTEMP}

suppress_history: false

windows:
  - window_name: vtr_navigation_system
    layout: main-horizontal
    shell_command_before:
      - source ${VTRSRC}/main/install/setup.bash
    panes:
      - >
        ros2 launch vtr_navigation vtr.launch.py
        base_params:=hdl64_grizzly_default.yaml
        data_dir:=${VTRTEMP}/lidar/
        start_new_graph:=true
        use_sim_time:=true

  - window_name: vtr_gui
    layout: main-horizontal
    shell_command_before:
      - source ${VTRSRC}/main/install/setup.bash
    panes:
      - ros2 run vtr_gui socket_client  --ros-args -r __ns:=/vtr
      - ros2 run vtr_gui socket_server  --ros-args -r __ns:=/vtr
      - ros2 run vtr_gui web_server  --ros-args -r __ns:=/vtr
      # - firefox --new-window "localhost:5200" # the webpage has to wait for everything above

  - window_name: rviz2
    layout: main-horizontal
    shell_command_before:
      - source ${VTRSRC}/main/install/setup.bash
    panes:
      - ros2 run rviz2 rviz2 -d ${VTRSRC}/rviz/hdl64.rviz
      # - ros2 run rqt_reconfigure rqt_reconfigure
